sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/data/MetricData.java

```diff
@@ -170,6 +171,16 @@ static MetricData createExponentialHistogram(
         data);
   }
 
+  /**
+   * Get an empty instance of metric data. Empty metric data should not be used except as an
+   * indication that recordings were dropped via {@link Aggregation#drop()}.
+   *
+   * @return an empty MetricData.
+   */
+  static MetricData empty() {
```

Comment:
Should we leave this out of public API until supporting custom aggregations?

Reply:
The Drop aggregator is exposed in the SDK, so it's possible to see this given this PR.


Second question though - Can you add some tests where this shows up in Exporters?  I'm curious if they need to be updated for this data type, I suspect it will cause breakages.

Reply:
> Second question though - Can you add some tests where this shows up in Exporters? I'm curious if they need to be updated for this data type, I suspect it will cause breakages.

The way I handled this was with a `MetricData.isEmpty()` check in `MeterSharedState#collectAll(...)`, and `EmptyMetricData.isEmpty()` always returns `true`:
```
  public List<MetricData> collectAll(
      CollectionInfo collectionInfo,
      MeterProviderSharedState meterProviderSharedState,
      long epochNanos,
      boolean suppressSynchronousCollection) {
    Collection<MetricStorage> metrics = getMetricStorageRegistry().getMetrics();
    List<MetricData> result = new ArrayList<>(metrics.size());
    for (MetricStorage metric : metrics) {
      MetricData current =
          metric.collectAndReset(
              collectionInfo,
              meterProviderSharedState.getResource(),
              getInstrumentationLibraryInfo(),
              meterProviderSharedState.getStartEpochNanos(),
              epochNanos,
              suppressSynchronousCollection);
      // Ignore if the metric data doesn't have any data points, for example when aggregation is
      // Aggregation#drop()
      if (!current.isEmpty()) {
        result.add(current);
      }
    }
    return result;
  }
```

Reply:
Added a unit test in latest commit

Reply:
> Should we leave this out of public API until supporting custom aggregations?

Not sure if this was answered - does it work to leave this out of the public API for now?

Reply:
Possibly, yes. 

The `MetricData` interface includes `Data<?> getData()`, and the `Data<T extends PointData>` interface is package private. This means that any implementation of `MetricData` has to live in `io.opentelemetry.sdk.metrics.data`. The fact that `MetricData` has a public API that returns a package private interface is probably a bug. 

But while its true that `MetricData` implementations needs to be in `io.opentelemetry.sdk.metrics.data`, there's no way (short of reflection) to have a `EmptyMetricData` implementation without some sort of public accessor, since a public accessor is needed for `DropAggregator` and other internal accesses. 

But if we make `Data` public, we can move `EmptyMetricData` to an internal package and remove `MetricData.empty()`.

Reply:
> The fact that MetricData has a public API that returns a package private interface is probably a bug.

Nice catch! Sounds like a plan